example/lib/develop/peripheral_list/peripheral_list_screen.dart

```diff
@@ -0,0 +1,129 @@
+        actions: <Widget>[
+          MaterialButton(
+            child: Text('Start scanning'),
+            onPressed: () => _startScanning(context),
+          ),
+          MaterialButton(
+            child: Text('Stop scanning'),
+            onPressed: () => _stopScanning(context),
+          ),
+        ],
+      ),
+      body: Container(
+        child: BlocBuilder<PeripheralListBloc, PeripheralListState>(
+            builder: (context, state) {
+          return ListView.builder(
+            itemCount: state.peripherals.length,
+            itemBuilder: (context, index) {
+              return _buildRow(context, state.peripherals[index]);
+            },
+            padding: EdgeInsets.fromLTRB(16.0, 8.0, 16.0, 8.0),
+          );
+        }),
+      ),
+      backgroundColor: CustomColors.systemGroupedBackground,
+    );
+  }
+
+  static Widget _buildRow(BuildContext context, BlePeripheral peripheral) {
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(0, 8.0, 0, 8.0),
+      child: ClipRRect(
+        borderRadius: BorderRadius.circular(8.0),
+        child: Container(
+          color: CustomColors.rowBackground,
+          child: ListTile(
+            leading: _buildListTileLeading(context),
+            title: Text(peripheral.name),
+            subtitle: Text(
+              peripheral.id,
+              maxLines: 1,
+            ),
+            trailing: _buildListTileTrailing(context, peripheral),
+            dense: true,
+          ),
+        ),
+      ),
+    );
+  }
+
+  static Widget _buildListTileLeading(BuildContext context) {
+    return CircleAvatar(
+      child: Icon(Icons.bluetooth),
+    );
+  }
+
+  static Widget _buildListTileTrailing(
+      BuildContext context, BlePeripheral peripheral) {
+    if (Platform.isIOS) {
+      return Row(
+        children: <Widget>[
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              ConnectionStateView(
+                isConnected: peripheral.isConnected,
+              ),
+              SizedBox(height: 4.0),
+              RssiView(
+                rssiValue: peripheral.rssiString,
+              ),
+            ],
+          ),
+          Icon(Icons.chevron_right),
+        ],
+        mainAxisSize: MainAxisSize.min,
+      );
+    } else {
+      return Row(
+        children: <Widget>[
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              ConnectionStateView(
+                isConnected: peripheral.isConnected,
+              ),
+              SizedBox(height: 4.0),
+              RssiView(
+                rssiValue: peripheral.rssiString,
+              ),
+            ],
+          ),
+        ],
+        mainAxisSize: MainAxisSize.min,
+      );
+    }
+  }
+
+  void _startScanning(BuildContext context) {
+    final peripheralListBloc = BlocProvider.of<PeripheralListBloc>(context);
```

Comment:
what about creating it once instead of creating each time a user taps on the button?

Reply:
Fixed - got the bloc from provider in the build method and passing it as a parameter to the button's onPressed method